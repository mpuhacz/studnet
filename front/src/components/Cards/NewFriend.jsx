import React from 'react';

const NewFriend = ({ name, avatar, likes}) => (
  <div className="event">
    <div className="label">
      <img src={avatar} alt="avatar" />
    </div>
    <div className="content">
      <div className="summary">
        <a className="user">
          {name}
        </a> dodał Cie jako znajomego
        <div className="date">
          1 godzinę temu
        </div>
      </div>
      <div className="meta">
        <a className="like">
          <i className="like icon"></i> {likes} Likes
        </a>
      </div>
    </div>
  </div>
);

export default NewFriend;
