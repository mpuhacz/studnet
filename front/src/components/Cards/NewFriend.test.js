import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import NewFriend from './NewFriend';

describe('<NewFriend />', () => {
  it('renders NewFriend without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NewFriend
      avatar="https://semantic-ui.com/images/avatar/small/justen.jpg"
      name="Name"
      likes={1}
    />, div);
  });

  it('renders NewFriend form with correct specification', () => {
    const wrapper = renderer.create(<NewFriend
      avatar="https://semantic-ui.com/images/avatar/small/justen.jpg"
      name="Name"
      likes={1}
    />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
