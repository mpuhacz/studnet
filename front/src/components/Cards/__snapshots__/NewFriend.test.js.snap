exports[`<NewFriend /> renders NewFriend form with correct specification 1`] = `
<div
  className="event">
  <div
    className="label">
    <img
      alt="avatar"
      src="https://semantic-ui.com/images/avatar/small/justen.jpg" />
  </div>
  <div
    className="content">
    <div
      className="summary">
      <a
        className="user">
        Name
      </a>
       dodał Cie jako znajomego
      <div
        className="date">
        1 godzinę temu
      </div>
    </div>
    <div
      className="meta">
      <a
        className="like">
        <i
          className="like icon" />
         
        1
         Likes
      </a>
    </div>
  </div>
</div>
`;
